distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Data_Link_Setup.enso

```diff
@@ -31,12 +33,18 @@ type Data_Link_Setup
                         _ -> Error.throw (Illegal_Argument.Error "Invalid value for `on_existing_file` parameter, only `Overwrite` and `Error` are supported here.")
                     exists_checked = if replace_existing.not && destination.exists then Error.throw (File_Error.Already_Exists destination)
                     exists_checked.if_not_error <|
-                        json = create_fn destination
+                        connection_description = create_fn destination
+                        json = link_type.add_to_data_link_description connection_description
                         Data_Link.write_config destination json replace_existing
                 _ -> Error.throw (Illegal_Argument.Error "Currently a connection can only be saved as a Data Link into the Enso Cloud. Please provide an `Enso_File` as destination.")
 
         Data_Link_Setup.Unavailable cause ->
-            Error.throw (Illegal_Argument.Error "Cannot save connection as Data Link: "+cause)
+            word = case link_type of
+                DB_Data_Link_Type.Database -> "connection"
+                DB_Data_Link_Type.Table _ -> "table"
+                DB_Data_Link_Type.Query _ -> "query"
+                DB_Data_Link_Type.SQL_Statement _ -> "query"
```

Comment:
method on DB_Data_Link?

Reply:
not sure what this method should be called, it's very context-dependent how this conversion is handled.